manifest/v1alpha/slo/validation.go

```diff
@@ -184,6 +260,13 @@ var specCompositeObjectiveValidationRule = validation.NewSingleRule(func(s Spec)
 	return nil
 })
 
+var compositeObjectiveRule = validation.New[CompositeObjective](
+	validation.ForPointer(func(c CompositeObjective) *float64 { return c.Weight }).
+		WithName("weight").
+		Required().
+		Rules(validation.GreaterThan(0.0)),
```

Comment:
`whenDelayed` is an enum so it should be validated against allowed set of values.

Reply:
Added, with test